xieyu_wang/Graph.cpp

```diff
@@ -56,7 +56,11 @@ bool Graph::isConnected(int fromVertexId, int toVertexId) {
   assert(hasVertex(toVertexId) && "Vertex doesn't exist");
   for (const auto& fromVertexEdgeId : getVertex(fromVertexId).getEdgeIds()) {
     for (const auto& toVertexEdgeId : getVertex(toVertexId).getEdgeIds()) {
-      if (fromVertexEdgeId == toVertexEdgeId && fromVertexId != toVertexId) {
+      if (fromVertexEdgeId == toVertexEdgeId) {
+        const auto edge = getEdge(fromVertexEdgeId);
+        if (edge.fromVertexId != edge.toVertexId) {
+          continue;
+        }
         return true;
```

Comment:
1) Why did you write it so complicated
```suggestion
        if (edge.fromVertexId == edge.toVertexId) {
          return true
        }
```
2. This is incorrect, it will only work for green edges, how's about other edges?

Look. You can check this small example:
```cpp
graph.addVertex();
graph.addVertex();
graph.addVertex();

graph.addEdge(0, 1); // gray
graph.addEdge(1, 2); // gray
graph.addEdge(1, 1); // green

assert(graph.isConnected(0, 1) == true); // gray
assert(graph.isConnected(1, 0) == true); // gray

assert(graph.isConnected(1, 2) == true); // gray
assert(graph.isConnected(2, 1) == true); // gray

assert(graph.isConnected(1, 1) == true); // green

assert(graph.isConnected(0, 0) == false); // no edge
assert(graph.isConnected(2, 2) == false); // no edge
assert(graph.isConnected(0, 2) == false); // no edge
assert(graph.isConnected(2, 0) == false); // no edge
```

This is how your graph should operate. Right now this function is broken.